src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs

```diff
@@ -305,10 +379,55 @@ await pipeline.ExecuteAsync(async (ctx) =>
                 {
                     resourceLogger.LogError("Could not import Redis databases into RedisInsight. Reason: {reason}", ex.Message);
                 }
-            };
+            }
         }
     }
 
+    /// <summary>
+    /// Initializes the Redis Insight settings to work around https://github.com/RedisInsight/RedisInsight/issues/3452.
+    /// Redis Insight requires the encryption property to be set if the Redis database connection contains a password.
+    /// </summary>
+    private static async Task InitializeRedisInsightSettings(HttpClient client, ILogger resourceLogger, CancellationToken ct)
+    {
+        if (await AreSettingsInitialized(client, ct).ConfigureAwait(false))
+        {
+            return;
+        }
+
+        var jsonContent = JsonContent.Create(new
+        {
+            agreements = new
+            {
+                // all 4 are required to be set
+                eula = false,
```

Comment:
This is unfortunate, but it is the only way I could get it to work. See https://github.com/RedisInsight/RedisInsight/issues/3452 for more information on why this is needed.

Note that this isn't accepting the EULA (it is sending false). But the consequences of initializing this is the pop up no longer happens on first run of Redis Insight, which is a nice user experience.

Thoughts? @sebastienros @DamianEdwards @Alirexaa

Reply:
What are the options?

Reply:
Seems [they're suggesting we wait](https://github.com/RedisInsight/RedisInsight/issues/3452#issuecomment-2623640956) until they implement support for adding databases via environment variables instead.

Reply:
I've opened [[Bug]: Posting eula: false to /api/settings suppresses the first run dialog (RedisInsight/RedisInsight#4340)](https://github.com/RedisInsight/RedisInsight/issues/4340) for the first run dialog issue.

Let's move forward with this PR for now. We can readjust later.